driver/display/ssd1306.zig

```diff
@@ -0,0 +1,855 @@
+        const page: u3 = @truncate(y / 8);
+        const bit: u3 = @truncate(y % 8);
+        const mask: u8 = @as(u8, 1) << bit;
+
+        const offset: usize = (@as(usize, page) << 7) + x;
+
+        switch (color) {
+            .black => fb.pixel_data[offset] &= ~mask,
+            .white => fb.pixel_data[offset] |= mask,
+        }
+    }
+};
+
+const ControlByte = packed struct(u8) {
+    zero: u6 = 0,
+
+    /// The D/C# bit determines the next data byte is acted as a command or a data. If the D/C# bit is
+    /// set to logic “0”, it defines the following data byte as a command. If the D/C# bit is set to
+    /// logic “1”, it defines the following data byte as a data which will be stored at the GDDRAM.
+    /// The GDDRAM column address pointer will be increased by one automatically after each
+    /// data write.
+    mode: enum(u1) { command = 0, data = 1 },
+
+    /// If the Co bit is set as logic “0”, the transmission of the following information will contain data bytes only.
+    co_bit: u1,
+
+    const command: u8 = @bitCast(ControlByte{
+        .mode = .command,
+        .co_bit = 0,
+    });
+
+    const data_byte: u8 = @bitCast(ControlByte{
+        .mode = .data,
+        .co_bit = 1,
+    });
+
+    const data_stream: u8 = @bitCast(ControlByte{
+        .mode = .data,
+        .co_bit = 0,
+    });
+};
+
+comptime {
+    std.debug.assert(ControlByte.command == 0x00);
+    std.debug.assert(ControlByte.data_byte == 0xC0);
+    std.debug.assert(ControlByte.data_stream == 0x40);
+}
+
+// Fundamental Commands
+const DisplayOnMode = enum(u8) { resumeToRam = 0xA4, ignoreRam = 0xA5 };
+const NormalOrInverseDisplay = enum(u8) { normal = 0xA6, inverse = 0xA7 };
+const DisplayMode = enum(u8) { off = 0xAE, on = 0xAF };
+
+// Scrolling Commands
+const HorizontalScrollDirection = enum(u8) { right = 0x26, left = 0x27 };
+const VerticalAndHorizontalScrollDirection = enum(u8) { right = 0x29, left = 0x2A };
+const PageError = error{EndPageIsSmallerThanStartPage};
+
+// Addressing Setting Commands
+const Column = enum(u1) { lower = 0, higher = 1 };
+const MemoryAddressingMode = enum(u2) { horizontal = 0b00, vertical = 0b01, page = 0b10 };
+
+// Hardware Configuration Commands
+const InputError = error{InvalidEntry};
+
+// Tests
+
+const TestDevice = mdf.base.DatagramDevice.TestDevice;
+
+// This is the command sequence created by SSD1306.init()
+// to set up the display.
+const recorded_init_sequence = [_][]const u8{
+    &.{ 0x00, 0xAE },
+    &.{ 0x00, 0x2E },
+    &.{ 0x00, 0xA1 },
+    &.{ 0x00, 0xC8 },
+    &.{ 0x00, 0xA6 },
+    &.{ 0x00, 0x81, 0xFF },
+    &.{ 0x00, 0x8D, 0x14 },
+    &.{ 0x00, 0xA8, 0x3F },
+    &.{ 0x00, 0xD5, 0x80 },
+    &.{ 0x00, 0xD9, 0x11 },
+    &.{ 0x00, 0xDB, 0x40 },
+    &.{ 0x00, 0xDA, 0x12 },
+    &.{ 0x00, 0xD3, 0x00 },
+    &.{ 0x00, 0x40 },
+    &.{ 0x00, 0xA4 },
+    &.{ 0x00, 0xAF },
+};
+
+// Fundamental Commands
+test "setContrast" {
```

Comment:
I think this file would be a great candidate for symbol tests. If not, then please rename the tests to match the snake case function names.